extmod/modopenamp.c

```diff
@@ -0,0 +1,394 @@
+    if (args[ARG_dst_addr].u_int != -1) {
+        dst_addr = args[ARG_dst_addr].u_int;
+    }
+
+    mp_buffer_info_t rbuf;
+    mp_get_buffer_raise(pos_args[1], &rbuf, MP_BUFFER_READ);
+    debug_printf("rpmsg_send() msg len: %d\n", rbuf.len);
+
+    int bytes = 0;
+    uint32_t timeout = args[ARG_timeout].u_int;
+    for (mp_uint_t start = mp_hal_ticks_ms(); ; mp_hal_delay_ms(10)) {
+        bytes = rpmsg_send_offchannel_raw(&self->ep, src_addr, dst_addr, rbuf.buf, rbuf.len, false);
+        if (bytes > 0) {
+            break;
+        }
+        if (timeout > 0 && (mp_hal_ticks_ms() - start > timeout)) {
+            mp_raise_msg(&mp_type_OSError, MP_ERROR_TEXT("timeout waiting for a free buffer"));
+        }
+        MICROPY_EVENT_POLL_HOOK
+    }
+    return mp_const_none;
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_KW(rpmsg_send_obj, 2, _rpmsg_send);
+
+STATIC mp_obj_t rpmsg_is_ready(mp_obj_t self_in) {
+    rpmsg_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    return is_rpmsg_ept_ready(&self->ep) ? mp_const_true : mp_const_false;
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_1(rpmsg_is_ready_obj, rpmsg_is_ready);
+
+STATIC mp_obj_t rpmsg_deinit(mp_obj_t self_in) {
+    rpmsg_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    rpmsg_destroy_ept(&self->ep);
+    return mp_const_none;
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_1(rpmsg_deinit_obj, rpmsg_deinit);
+
+STATIC mp_obj_t rpmsg_make_new(const mp_obj_type_t *type, size_t n_args, size_t n_kw, const mp_obj_t *all_args) {
+    enum { ARG_name, ARG_src_addr, ARG_dst_addr, ARG_callback };
+    static const mp_arg_t allowed_args[] = {
+        { MP_QSTR_name, MP_ARG_OBJ | MP_ARG_REQUIRED,  {.u_rom_obj = MP_ROM_NONE } },
+        { MP_QSTR_src_addr, MP_ARG_INT | MP_ARG_KW_ONLY,  {.u_int = RPMSG_ADDR_ANY } },
+        { MP_QSTR_dst_addr, MP_ARG_INT | MP_ARG_KW_ONLY,  {.u_int = RPMSG_ADDR_ANY } },
+        { MP_QSTR_callback, MP_ARG_OBJ,  {.u_rom_obj = MP_ROM_NONE } },
+    };
+
+    // Parse args.
+    mp_arg_val_t args[MP_ARRAY_SIZE(allowed_args)];
+    mp_arg_parse_all_kw_array(n_args, n_kw, all_args, MP_ARRAY_SIZE(allowed_args), allowed_args, args);
+
+    rpmsg_obj_t *self = mp_obj_malloc_with_finaliser(rpmsg_obj_t, &rpmsg_type);
+    self->name = args[ARG_name].u_obj;
+    self->callback = args[ARG_callback].u_obj;
+
+    if (rpmsg_create_ept(&self->ep, &MP_STATE_PORT(virtio_device)->rvdev.rdev, mp_obj_str_get_str(self->name),
+        args[ARG_src_addr].u_int, args[ARG_dst_addr].u_int, rpmsg_recv_callback, NULL) != 0) {
+        mp_raise_msg(&mp_type_OSError, MP_ERROR_TEXT("Failed to create RPMsg endpoint"));
+    }
+
+    return MP_OBJ_FROM_PTR(self);
+}
+
+STATIC const mp_rom_map_elem_t rpmsg_locals_dict_table[] = {
+    { MP_ROM_QSTR(MP_QSTR___name__), MP_ROM_QSTR(MP_QSTR_rpmsg) },
+    { MP_ROM_QSTR(MP_QSTR___del__), MP_ROM_PTR(&rpmsg_deinit_obj) },
+    { MP_ROM_QSTR(MP_QSTR_send), MP_ROM_PTR(&rpmsg_send_obj) },
+    { MP_ROM_QSTR(MP_QSTR_is_ready), MP_ROM_PTR(&rpmsg_is_ready_obj) },
```

Comment:
Although I prefer `is_ready` too, the convention elsewhere in MicroPython would be `isready`. (e.g. wlan.isconnected()`.

Reply:
I will change it if we must, but I'd argue that it's `isconnected` that's wrong and should be made obsolete, with a non-breaking change by just adding an alias `is_connected`. Not in MicroPython there's `i2c.is_ready` and `is_connected` in bluetooth stuff.
 CC @dpgeorge

Reply:
> and `is_connected` in bluetooth stuff

Where is this exactly?

> I'd argue that it's `isconnected` that's wrong and should be made obsolete, with a non-breaking change by just adding an alias `is_connected`

Yes, we could do that.

Are there any other places where the convention "without underscore" is used that would also need changing?  Eg something like `onxxx`.

Reply:
> Where is this exactly?

In `micropython-lib` aioble.

> Are there any other places where the convention "without underscore" is used

It's hard to find them, as they match a lot of things, but math functions `isinf` etc.. I don't know if you want to change those or not.

Reply:
> In `micropython-lib` aioble.

Ah, I see, thanks.

> It's hard to find them, as they match a lot of things, but math functions `isinf` etc

They can't be changed, that's CPython (and ultimately C library) legacy.  (Although note that CPython has things like `pathlib.Path.is_dir`, so isn't fully consistent itself.)

I don't think in MicroPython's custom modules (eg `machine`, `network`) that we have many of these cases.  For example, in the `micropython` module all the functions there have `_` to separate words, eg `stack_use`, `heap_lock`.  There are things like `readinto` but they also have history from CPython.

If `isconnected()` is the only case that needs to be changed to make things semi-consistent then maybe it's worth doing (in a backwards compatible way), and then this here can be `is_ready`.

Reply:
>> If isconnected() is the only case that needs to be changed to make things

`isconnected()` is the only case I could find and it seems to be common in wlan modules. There's already `is_ready` but no `isready` so I think this case here should definitely stay `is_ready`